java/src/main/java/com/google/cloud/dataproc/templates/kafka/KafkaToBQDstream.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.spark.api.java.function.VoidFunction2;
+import org.apache.spark.sql.*;
+import org.apache.spark.streaming.Duration;
+import org.apache.spark.streaming.Time;
+import org.apache.spark.streaming.api.java.JavaInputDStream;
+import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.apache.spark.streaming.kafka010.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+/**
+ * Spark job to move data or/and schema from Kafka topic to BQ via spark Direct Stream. For detailed
+ * list of properties refer "KafkaToBQDstream Template properties" section in
+ * resources/template.properties file.
+ */
+public class KafkaToBQDstream implements BaseTemplate {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(KafkaToBQDstream.class);
+
+  private String projectId;
+  private long batchInterval;
+  private String bqWriteMode;
+  private String kafkaStartingOffsets;
+  private String kafkaGroupId;
+  private String kafkaBootstrapServers;
+  private String kafkaTopic;
+  private final String sparkLogLevel;
+  private Long kafkaAwaitTerminationTimeout;
+  private String bigQueryDataset;
+  private String bigQueryTable;
+  private String tempGcsBucket;
+
+  public KafkaToBQDstream() {
+
+    projectId = getProperties().getProperty(PROJECT_ID_PROP);
+    kafkaBootstrapServers = getProperties().getProperty(KAFKA_BOOTSTRAP_SERVERS);
+    kafkaTopic = getProperties().getProperty(KAFKA_TOPIC);
+    kafkaStartingOffsets = getProperties().getProperty(KAFKA_STARTING_OFFSET);
+    kafkaGroupId = getProperties().getProperty(KAFKA_BQ_CONSUMER_GROUP_ID);
+    batchInterval = Long.parseLong(getProperties().getProperty(KAFKA_BQ_BATCH_INTERVAL));
+    bqWriteMode = getProperties().getProperty(KAFKA_BQ_STREAM_OUTPUT_MODE);
+    sparkLogLevel = getProperties().getProperty(SPARK_LOG_LEVEL);
+    bigQueryDataset = getProperties().getProperty(KAFKA_BQ_DATASET);
+    bigQueryTable = getProperties().getProperty(KAFKA_BQ_TABLE);
+    tempGcsBucket = getProperties().getProperty(KAFKA_BQ_TEMP_GCS_BUCKET);
+    kafkaAwaitTerminationTimeout =
+        Long.valueOf(getProperties().getProperty(KAFKA_BQ_AWAIT_TERMINATION_TIMEOUT));
+  }
+
+  @Override
+  public void runTemplate() throws TimeoutException, SQLException, InterruptedException {
+
+    SparkSession spark =
+        SparkSession.builder().appName("Kafka to BQ via Direct stream").getOrCreate();
+
+    spark.sparkContext().setLogLevel(sparkLogLevel);
+
+    Map<String, Object> kafkaParams = new HashMap<>();
+    kafkaParams.put("bootstrap.servers", kafkaBootstrapServers);
+    kafkaParams.put("key.deserializer", StringDeserializer.class);
+    kafkaParams.put("value.deserializer", StringDeserializer.class);
+    kafkaParams.put("group.id", kafkaGroupId);
+    kafkaParams.put("auto.offset.reset", kafkaStartingOffsets);
+    kafkaParams.put("enable.auto.commit", false);
+
+    Collection<String> topics = Collections.singletonList(kafkaTopic);
```

Comment:
Can kafkaTopic include multiple topic list like: t1,t2,t3 ?

Reply:
Yes. As per KafkaUtils.createDirectStream ,  it is expecting  list of topics